internal/runtime/source.go

```diff
@@ -135,6 +138,13 @@ func ParseSources(sources map[string][]byte) (*Source, error) {
 
 		sourceFragment, err := ParseSource(namedSource.Name, namedSource.Content)
 		if err != nil {
+			// If we encounter diagnostic errors we combine them and
+			// later return all of them
+			var diags diag.Diagnostics
+			if errors.As(err, &diags) {
+				mergedDiags = append(mergedDiags, diags...)
+				continue
```

Comment:
This seems like we should continue whether or not we received diags, as we might not receive diags from the first source but could receive them from later sources. What is the case where we want to exit early on an error?

Reply:
Yes but I don't really now a good way to combine diagnostic and non diagnostic errors. Currently the only other error source would be from [EnsureUTF8](https://github.com/grafana/alloy/blob/8d5b1b4ef2f47a3db2f1116af4988c98766920c0/internal/runtime/source.go#L33). We could try to build a diagnostic error from this one but we don't have an parsed AST because this happens before we can try to parse it so I am not sure what we would set the start and end positions to.

Reply:
Fair enough, once you fix UTF-8 errors you can get all diags, good enough. 👍